packages/mana/consensusbasevector_test.go

```diff
@@ -653,3 +654,91 @@ func TestConsensusBaseManaVector_ToAndFromPersistable(t *testing.T) {
 	}
 	assert.Equal(t, bmv.(*ConsensusBaseManaVector).vector, restoredBmv.(*ConsensusBaseManaVector).vector)
 }
+
+func TestConsensusBaseManaVector_BuildPastBaseVector(t *testing.T) {
```

Comment:
Let's extend this test with new cases:
 - called with empty event log set,
 - called with time that is older than any of the logged events,
 - called with a time that causes some events not to be consumed in event log,
 - accidentally called with events that are already included in the vector, revoke should panic due to negative mana values.

Also, see below my comments on the constructed `txInfo`s.